packages/wdio-appium-service/src/launcher.js

```diff
@@ -44,6 +52,19 @@ export default class AppiumLauncher {
             this.command = 'cmd'
         }
 
+        /**
+         * update capability connection options to connect
+         * to Appium server
+         */
+        (
+            Array.isArray(this.capabilities)
+                ? this.capabilities
+                : Object.values(this.capabilities)
```

Comment:
> By doing this you lose the object reference it it will not be mapped to this.capabilities

It will be mapped (otherwise the unit test will fail). Note that in JavaScript variables are pointing to the actual object in memory:

```js
const foo = { a: 1, b: 2 }
const bar = foo
bar.a = 3
console.log(foo.a) // outputs 3
```

Reply:
> Maybe also mention why we do the check here (multi remote)

What do you mean by that?

Reply:
the capabilities are an array when not multi remote but an object when multi remote, right?